packages/grafana-ui/src/components/Table/TableNG/Cells/renderers.tsx

```diff
@@ -118,17 +153,24 @@ const CELL_REGISTRY: Record<TableCellOptions['type'], CellRegistryEntry> = {
       />
     )),
     getStyles: getPillStyles,
-    testField: (field: Field) =>
-      field.type === FieldType.string ||
-      (field.type === FieldType.other && field.values.some((val) => Array.isArray(val))),
+    testField: (field: Field) => {
+      if (PRIMITIVE_TYPES.includes(field.type)) {
+        return true;
+      }
+      if (field.type === FieldType.other) {
+        const firstNonNullValue = field.values.find((v) => v != null);
+        return Array.isArray(firstNonNullValue);
+      }
+      return false;
+    },
   },
   [TableCellDisplayMode.Markdown]: {
     // eslint-disable-next-line react/display-name
     renderer: memo((props: TableCellRendererProps) => (
       <MarkdownCell field={props.field} rowIdx={props.rowIdx} disableSanitizeHtml={props.disableSanitizeHtml} />
     )),
     getStyles: getMarkdownCellStyles,
-    testField: (field: Field) => field.type === FieldType.string,
+    testField: (field: Field) => PRIMITIVE_TYPES.includes(field.type),
```

Comment:
I made Markdown and Pill cell a little more permissive in this PR, given some ad hoc feedback I've gotten on Pill cell where people wanted to render a number in there for whatever reason. I can see a situation where you'd want to set the whole table to Markdown and pass primitive types through rather than set up field overrides, for example.